### PR TITLE
[NUI][Xaml] Fix issue that BindingContext can't be used in C# code

### PR DESCRIPTION
**Pull request by AdunFang**
### Description of Change ###

1. Fix issue that BindingContext can't effect the child of child.
2. Fix issue that BindingContext can't effect the binding which setted by C# code.

### API Changes ###

 - ACR:

### CODE REVIEW

**Review by elishateng — APPROVED** [review]
LGTM

**Review by huayongxu — APPROVED** [review]
lgtm.

**Review by zhouleonlei — APPROVED** [review]
LGTM

File: src/Tizen.NUI/src/public/XamlBinding/BindableObject.cs [hunk 1]
```diff
@@ -44,15 +44,36 @@ public abstract class BindableObject : INotifyPropertyChanged, IDynamicResourceH
...
         /// <summary>
         /// Gets or sets object that contains the properties that will be targeted by the bound properties that belong to this BindableObject.
         /// </summary>
         /// This will be public opened in tizen_5.0 after ACR done. Before ACR, need to be hidden as inhouse API.
         [EditorBrowsable(EditorBrowsableState.Never)]
         public object BindingContext
         {
-            get { return inheritedContext ?? GetValue(BindingContextProperty); }
-            set { SetValue(BindingContextProperty, value); }
+            get
```
Comment (myroot) [inline src/Tizen.NUI/src/public/XamlBinding/BindableObject.cs:56]: Why don't you use `BindingContextProperty`? `BindingContext` also BindableProperty, so it should be implemented with `BindableProperty` to notify changes of value and to support binding to value ....

**Comment by dongsug-song** [comment]
@AdunFang 
please check this VS warning also.
=>
src/Tizen.NUI/src/public/XamlBinding/BindableObject.cs#L383

In virtual/interface member BindableObject.OnPropertyChangedWithData(BindableProperty), rename parameter property so that it no longer conflicts with the reserved language keyword 'Property'. Using a reserved keyword as the name of a parameter on a virtual/interface member makes it harder for consumers in other languages to override/implement the member

File: src/Tizen.NUI/src/public/XamlBinding/BindableObject.cs (hunk 1 above)
Comment (AdunFang) [reply src/Tizen.NUI/src/public/XamlBinding/BindableObject.cs]: Fixed it.

**Comment by dongsug-song** [comment]
same PR ( https://github.com/Samsung/TizenFX/pull/3132 ) has been merged, so I close this now.

**Comment by myroot** [comment]
The original implement of `BindableObject` is support a chain of BindingContext between parent and descendant. 
I think, you do better to fix it to work rather than this PR. 

As I search, 
https://github.com/Samsung/TizenFX/blob/6405fb6c273725b83ad82ad5e237bd109b314f6b/src/Tizen.NUI/src/public/XamlBinding/Element.cs#L324-L349
This code propagate `BindingContext` to descendant, but Any NUI classes are not override `LogicalChildrenInternal`

**Pull request closed without merging**